integrationTests/vm/arwen/arwenvm/mandosConverter/mandosConverter.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/state"
+	"github.com/ElrondNetwork/elrond-go/testscommon/txDataBuilder"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+var errReturnCodeNotOk = errors.New("returnCode is not 0(Ok)")
+
+// CreateAccountsFromMandosAccs uses mandosAccounts to populate the AccountsAdapter
+func CreateAccountsFromMandosAccs(tc *vm.VMTestContext, mandosUserAccounts []*mge.TestAccount) (err error) {
+	for _, mandosAcc := range mandosUserAccounts {
+		acc, err := tc.Accounts.LoadAccount(mandosAcc.GetAddress())
+		if err != nil {
+			return err
+		}
+		account := acc.(state.UserAccountHandler)
+		account.IncreaseNonce(mandosAcc.GetNonce())
+		err = account.AddToBalance(mandosAcc.GetBalance())
+		if err != nil {
+			return err
+		}
+
+		mandosAccStorage := mandosAcc.GetStorage()
+		for key, value := range mandosAccStorage {
+			err = account.DataTrieTracker().SaveKeyValue([]byte(key), value)
+			if err != nil {
+				return err
+			}
+		}
+
+		accountCode := mandosAcc.GetCode()
+		if len(accountCode) != 0 {
+			account.SetCode(accountCode)
+			ownerAddress := mandosAcc.GetOwner()
+			account.SetOwnerAddress(ownerAddress)
+			account.SetCodeMetadata([]byte{0, 0})
+		}
+		err = tc.Accounts.SaveAccount(account)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = tc.Accounts.Commit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateTransactionsFromMandosTxs converts mandos transactions intro trasnsactions that can be processed by the txProcessor
+func CreateTransactionsFromMandosTxs(mandosTxs []*mge.Transaction) (transactions []*transaction.Transaction) {
+	transactions = make([]*transaction.Transaction, 0)
+	for _, mandosTx := range mandosTxs {
+		gasLimit, gasPrice := mandosTx.GetGasLimitAndPrice()
+		var data []byte
+		esdtTransfers := mandosTx.GetESDTTransfers()
+		endpointName := mandosTx.GetCallFunction()
+		args := mandosTx.GetCallArguments()
+		if len(esdtTransfers) != 0 {
+			data = mgutil.CreateMultiTransferData(mandosTx.GetReceiverAddress(), esdtTransfers, endpointName, args)
+		} else {
+			data = createData(endpointName, args)
+		}
+
+		tx := vm.CreateTransaction(
+			mandosTx.GetNonce(),
+			mandosTx.GetCallValue(),
+			mandosTx.GetSenderAddress(),
+			mandosTx.GetReceiverAddress(),
+			gasPrice,
+			gasLimit,
+			data)
+		if len(esdtTransfers) != 0 {
+			tx.RcvAddr = tx.SndAddr
+		}
+		transactions = append(transactions, tx)
+	}
+	return transactions
+}
+
+// DeploySCsFromMandosDeployTxs deploys all smartContracts correspondent to "scDeploy" in a mandos test, then replaces with the correct computed address in all the transactions.
+func DeploySCsFromMandosDeployTxs(testContext *vm.VMTestContext, deployMandosTxs []*mge.Transaction, mandosTxs []*mge.Transaction, deployedScAccounts []*mge.TestAccount) (err error) {
```

Comment:
Consider splitting this function in two, separating the tasks of (1) deploying the SCs and collecting their new addresses and then (2) injecting the new addresses in the `Mandos` transactions that need to call the newly deployed SCs. As it is, the function is somewhat opaque and it's not clear whether it does what the author intended.